src/request.js

```diff
@@ -100,12 +100,13 @@ function handleMessage(rawMessage, response) {
 	if (!observerObj) throw new Error('No associated request for the server message', rawMessage);
 	else {
 		let observer = observerObj.observer;
+		const request = observerObj.request;
 		response.body.__header = response.header;
 
 		clearTimeout(observerObj.timeout);
 
 		if (response.header.statusCode !== 200) {
-			observer.onError(response.body);
+			observer.onError(response.body, request);
```

Comment:
For consistency sake, should we add the request and response to the other middleware as well?